File: Platform.App/node_template/app/queryService.js

/**
 * Executa uma query 
 */
class QueryService{

    constructor(){}

    /**
     * @method filter 
     * @param {*} appId 
     * @param {*} mappedEntity 
     * @param {*} entity 
     * @param {*} projection 
     * @param {*} validityPolicy 
     * @description recupera uma instância de uma entidade de uma aplicação
     */
    filter(appId, mappedEntity, entity, projection, validityPolicy){        
        return new Promise(
            (resolve, reject) => {

                validityPolicy.setQueryContext(appId,mappedEntity,entity);
                validityPolicy.query(
                    projection, 
                    (result)=>{
                        resolve(result);
                    }, 
                    (e)=>{
                        reject(e);
                    }
                );
            }
        );
    }
    
}

module.exports = QueryService;

File: Platform.App/python-template/app/query_service.py
from model.query import Query


class QueryService:

    def __init__(self, reference_date, version, session):
        self.reference_date = reference_date
        self.version = version
        self.session = session

    def filter(self, app_id, mapped_entity, entity, projection):
        """ execute a query on domain """
        query = Query(self.reference_date, self.version, self.session)
        query.set_query_context(app_id, mapped_entity, entity)
        return query.execute(projection)
